stack/stack.rb

```diff
@@ -0,0 +1,35 @@
+class Stack
+  def initialize(size)
+    @size = size
+    @store = Array.new(@size)
+    @top = -1
+  end
+
+  def push(element)
+    if full?
+      raise "overflow error"
+    else
+      @store[@top] = element
+    end
+    @top += 1
+  end
+
+  def pop
+    if empty? or @store.nil?
+      p "empty"
+    else
+      @store.pop
+    end
+    @top -= 1
+  end
+
+
+  private
+    def empty?
```

Comment:
spaces are required here?

Reply:
@KaveriMoger we should not override ruby build in functionalities(empty?) for a specific purpose, either we should define it for generic purpose or should rename it.